common/src/main/java/tachyon/collections/IndexedSet.java

```diff
@@ -169,24 +170,36 @@ public void clear() {
    * {@code (o == null ? o2 == null : o.equals(o2))}. If this set already contains the object, the
    * call leaves the set unchanged.
    *
-   * @param object the object to add
+   * @param objToAdd the object to add
    * @return true if the object is successfully added to all indexes, otherwise false
    */
-  public boolean add(T object) {
-    Preconditions.checkNotNull(object);
+  public boolean add(T objToAdd) {
+    Preconditions.checkNotNull(objToAdd);
     synchronized (mLock) {
-      boolean success = mObjects.add(object);
+      boolean success = mObjects.add(objToAdd);
+      // iterate over the first level index (filedid to map)
       if (success) {
         for (Map.Entry<FieldIndex<T>, Integer> index : mIndexMap.entrySet()) {
+          // get the second level map
           Map<Object, Set<T>> fieldValueToSet = mSetIndexedByFieldValue.get(index.getValue());
-          Object value = index.getKey().getFieldValue(object);
-          if (fieldValueToSet.containsKey(value)) {
-            success = fieldValueToSet.get(value).add(object) && success;
+          // retrieve the key to index the object within the second level map
+          Object fieldValue = index.getKey().getFieldValue(objToAdd);
+          if (fieldValueToSet.containsKey(fieldValue)) {
+            success = fieldValueToSet.get(fieldValue).add(objToAdd);
+            if (!success) {
+              // this call can never return false because:
+              //   a. toAddSet is an HashSet of unbounded space
```

Comment:
Instead of camel case, can you just phrase "the set to add", or "fieldValueToSet.get(fieldValue)"?